packages/contracts-bedrock/test/properties/medusa/fuzz/Protocol.guided.t.sol

```diff
@@ -73,14 +75,86 @@ contract ProtocolGuided is ProtocolHandler {
+        OptimismSuperchainERC20 sourceToken = OptimismSuperchainERC20(allSuperTokens[fromIndex]);
+        OptimismSuperchainERC20 destinationToken =
+            MESSENGER.crossChainMessageReceiver(address(sourceToken), destinationChainId);
+        bytes32 deploySalt = MESSENGER.superTokenInitDeploySalts(address(sourceToken));
+        uint256 sourceBalanceBefore = sourceToken.balanceOf(currentActor());
+        uint256 sourceSupplyBefore = sourceToken.totalSupply();
+
+        vm.prank(currentActor());
+        try sourceToken.sendERC20(recipient, amount, destinationChainId) {
+            (, uint256 currentlyInTransit) = ghost_tokensInTransit.tryGet(deploySalt);
+            ghost_tokensInTransit.set(deploySalt, currentlyInTransit + amount);
+            // 26
+            uint256 sourceBalanceAfter = sourceToken.balanceOf(currentActor());
+            assert(sourceBalanceBefore - amount == sourceBalanceAfter);
+            // 10
+            uint256 sourceSupplyAfter = sourceToken.totalSupply();
+            assert(sourceSupplyBefore - amount == sourceSupplyAfter);
+        } catch {
+            // 6
+            assert(address(destinationToken) == address(sourceToken) || sourceBalanceBefore < amount);
+        }
+    }
+
+    /// @custom:property-id 11
+    /// @custom:property relayERC20 increases the token's totalSupply in the destination chain exactly by the input amount
+    /// @custom:property-id 27
+    /// @custom:property relayERC20 increases sender's balance in the destination chain exactly by the input amount
+    /// @custom:property-id 7
+    /// @custom:property calls to relayERC20 always succeed as long as the cross-domain caller is valid
+    function fuzz_RelayERC20(uint256 messageIndex) external {
+        MockL2ToL2CrossDomainMessenger.CrossChainMessage memory messageToRelay = MESSENGER.messageQueue(messageIndex);
+        OptimismSuperchainERC20 destinationToken = OptimismSuperchainERC20(messageToRelay.crossDomainMessageSender);
+        uint256 destinationSupplyBefore = destinationToken.totalSupply();
+        uint256 destinationBalanceBefore = destinationToken.balanceOf(messageToRelay.recipient);
+
+        try MESSENGER.relayMessageFromQueue(messageIndex) {
+            bytes32 deploySalt = MESSENGER.superTokenInitDeploySalts(address(destinationToken));
+            (bool success, uint256 currentlyInTransit) = ghost_tokensInTransit.tryGet(deploySalt);
+            // if sendERC20 didnt intialize this, then test suite is broken
+            assert(success);
+            ghost_tokensInTransit.set(deploySalt, currentlyInTransit - messageToRelay.amount);
+            // 11
+            assert(destinationSupplyBefore + messageToRelay.amount == destinationToken.totalSupply());
+            // 27
+            assert(
+                destinationBalanceBefore + messageToRelay.amount == destinationToken.balanceOf(messageToRelay.recipient)
+            );
+        } catch {
+            // 7
+            assert(false);
```

Comment:
couldn't it fail when the length of relayed messages is `0` and then we need to check that if it failed was for that reason?

Reply:
in that case the revert will be internal to this contract and not easy to `try/catch`, but we can safely ignore it in that case and medusa will re-try more often the call combinations where relays happen after queueing messages, since that leads to more coverage